### PR TITLE
Derive macaroon_secret_key from signing key.

### PR DESCRIPTION
**Pull request by erikjohnston**
Unfortunately, there are people that are running synapse without a
`macaroon_sercret_key` set. Mandating they set one is a good solution,
except that breaking auto upgrades is annoying.

### CODE REVIEW

**Comment by illicitonion** [comment]
Fix the test please :)

**Comment by erikjohnston** [comment]
@illicitonion PTAL :)

File: tests/config/test_load.py [hunk 1]
```diff
@@ -54,10 +54,11 @@ def test_generates_and_loads_macaroon_secret_key(self):
                 "was: %r" % (config.macaroon_secret_key,)
             )
 
-    def test_load_fails_if_macaroon_secret_key_missing(self):
+    def test_load_suceeds_if_macaroon_secret_key_missing(self):
```
Comment (illicitonion) [inline tests/config/test_load.py]: typo: s/suceeds/succeeds/

**Comment by illicitonion** [comment]
LGTM aside from typo

File: tests/config/test_load.py (hunk 1 above)
Comment (erikjohnston) [inline tests/config/test_load.py]: *coughs*

**Pull request merged**